metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowExecutor.java

```diff
@@ -232,35 +247,51 @@ private void periodicCheckingLoop(long sleepTime, AbstractMetisPlugin abstractMe
-        if (consecutiveCancelOrMonitorFailures == MAX_CANCEL_OR_MONITOR_FAILURES) {
-          monitorResult = new MonitorResult(TaskState.DROPPED, e.getMessage());
-          break;
+        if (ExternalRequestUtil
+            .doesExceptionCauseMatchAnyOfProvidedExceptions(mapWithRetriableExceptions, e)) {
+          consecutiveCancelOrMonitorFailures++;
+          LOGGER.warn(String.format(
+              "Monitoring of external task failed %s. After exceeding %s retries, pending status will be set",
+              consecutiveCancelOrMonitorFailures, MAX_CANCEL_OR_MONITOR_FAILURES), e);
+          if (consecutiveCancelOrMonitorFailures == MAX_CANCEL_OR_MONITOR_FAILURES) {
+            //Set pending status once
+            abstractMetisPlugin.setPluginStatusAndResetFailMessage(PluginStatus.PENDING);
+          }
+        } else {
+          // Set plugin to FAILED and return immediately
+          abstractMetisPlugin.setFinishedDate(null);
+          abstractMetisPlugin.setPluginStatusAndResetFailMessage(PluginStatus.FAILED);
+          abstractMetisPlugin.setFailMessage(MONITOR_ERROR_PREFIX + e.getMessage());
+          return;
         }
+      } finally {
+        workflowExecutionDao.updateMonitorInformation(workflowExecution);
       }
     } while (monitorResult == null || (monitorResult.getTaskState() != TaskState.DROPPED
         && monitorResult.getTaskState() != TaskState.PROCESSED));
 
-    preparePluginStateAndFinishedDate(abstractMetisPlugin, monitorResult,
-        consecutiveCancelOrMonitorFailures == MAX_CANCEL_OR_MONITOR_FAILURES);
+    preparePluginStateAndFinishedDate(abstractMetisPlugin, monitorResult);
   }
 
   private boolean shouldPluginBeCancelled(AbstractMetisPlugin abstractMetisPlugin,
       AtomicInteger previousProcessedRecords,
       AtomicLong checkPointDateOfProcessedRecordsPeriodInMillis) {
     // A plugin with CLEANING state is NOT cancellable, it will be when the state is updated
-    return ((abstractMetisPlugin.getPluginStatus() != PluginStatus.CLEANING && workflowExecutionDao
-        .isCancelling(workflowExecution.getId())) || isMinuteCapOverWithoutChangeInProcessedRecords(
-        abstractMetisPlugin, previousProcessedRecords,
-        checkPointDateOfProcessedRecordsPeriodInMillis));
+    final boolean notCleaningAndCancelling =
+        abstractMetisPlugin.getPluginStatus() != PluginStatus.CLEANING && workflowExecutionDao
+            .isCancelling(workflowExecution.getId());
+    // A cleaning or a pending task should not be cancelled by exceeding the minute cap
+    final boolean notCleaningOrPendingAndMinuteCapExceeded =
+        (abstractMetisPlugin.getPluginStatus() != PluginStatus.CLEANING
+            && abstractMetisPlugin.getPluginStatus() != PluginStatus.PENDING)
+            && isMinuteCapOverWithoutChangeInProcessedRecords(abstractMetisPlugin,
+            previousProcessedRecords, checkPointDateOfProcessedRecordsPeriodInMillis);
```

Comment:
How do you prevent that a task is cancelled by the minute cap as soon as it exits the pending state and becomes running again?

Maybe solve that in isMinuteCapOverWithoutChangeInProcessedRecords() in the same way as with CLEANING?

Reply:
Fixed